src/components/SignInButton.tsx

```diff
@@ -173,27 +176,48 @@ export const SignInButton = (props: SignInButtonProps) => {
     );
   };
 
+  useEffect(() => {
+    const init = async () => {
+      setIsSamDown(!(await ServiceStatus.isSamHealthy()));
```

Comment:
Optional suggestion. I was thinking that the consent-not-healthy case is another one where the user should be precluded from signing in. Slightly out of scope for this ticket, but since we have `ServiceStatus` methods now, it might make sense to add that check here too.

Reply:
I'll create a separate ticket.